### ISSUE TITLE
Prediction from images

### ISSUE THREAD

**Issue by ogencoglu** [udibr/sketch#2]
Can this be used to predict what is written in an image? If so, how would I do that?

Thanks.

**Comment by udibr** [comment]
In theory yes. You can find code that takes an image, runs a convolutional deep neural network (CNN) which converts the image to a vector and then feeds this vector as the start point for a recursive neural network (RNN) which generates text. Once you have this system working you can take this code to replace the CNN part. You can take the hidden state of the RNN when the drawing is complete and feed it as the start point of the different RNN which generates text. My guess is that the original code with CNN  will give better results.

**Comment by ogencoglu** [comment]
Thanks for the reply. Quick question: Is this CNN + RNN scheme a one shot learner or is it more like a 2D convolutional autoencoder to learn a feature vector and another learning for RNN in 2 separate trainings?

**Comment by udibr** [comment]
from what I know, it is "one shot" but you start with a CNN which was trained to classify images (e.g. VGG16) and the transfer the CNN model (without the top softmax part) to your CNN+RNN network.
So usually the CNN is much better trained and much deeper than the code in this project which is an auto-encoder. This is why I think you will get much better results with the standard CNN+RNN

The RNN is also using pre-trained information  especially the word2vec vectors.

**Comment by ogencoglu** [comment]
Cool, thanks. I'll check.
